docs/conf.py

```diff
@@ -268,9 +268,20 @@ def update_html_context(
     context["generate_toctree_html"] = partial(context["generate_toctree_html"], startdepth=0)
 
 
-def setup(app: Sphinx) -> dict[str, bool]:
-    app.setup_extension("litestar_sphinx_theme")
+def delayed_setup(app: Sphinx) -> None:
+    # When running linkcheck pydata_sphinx_theme causes a build failure, and checking
+    # the builder in the initial `setup` function call is not possible, so the check
+    # and extension setup has to be delayed until the builder is initialized.
```

Comment:
Docstrings for consistency 
```suggestion
"""
When running `linkcheck` `pydata_sphinx_theme` causes a build failure, and checking  the builder in the initial `setup` function call is not possible, so the check and extension setup has to be delayed until the builder is initialized.
"""
```